src/style-spec/validate/validate_filter.js

```diff
@@ -7,13 +8,19 @@ import {unbundle, deepUnbundle} from '../util/unbundle_jsonlint.js';
 import extend from '../util/extend.js';
 import {isExpressionFilter} from '../feature_filter/index.js';
 
-export default function validateFilter(options) {
+import type {ValidationOptions} from './validate.js';
+
+type Options = ValidationOptions & {
+    layerType: string;
+}
+
+export default function validateFilter(options: Options): Array<ValidationError> {
     if (isExpressionFilter(deepUnbundle(options.value))) {
         const layerType = deepUnbundle(options.layerType);
```

Comment:
Given `options.layerType` is a string, would it be possible to skip this call to `deepUnbundle` and then skip the cast to string below?